Tests/mocks/devicesettings.h

```diff
@@ -1218,3 +1231,420 @@ class FrontPanelConfig {
 
 }
 
+
+typedef uint32_t Op_t;
+class Exception : public std::exception
+{
+	public:
+		virtual const char* what() const throw()
+		{
+			return "Base Exception..";
+		}
+
+};
+class CECNoAckException : public Exception
+{
+	public:
+		virtual const char* what() const throw()
+		{
+			return "Acl not received..";
+		}
+};
+class IOException : public Exception
+{
+	public:
+		virtual const char* what() const throw()
+		{
+			return "IO Exception..";
+		}
+};
+
+class CECFrameImpl{
+	public:
+		virtual void getBuffer(const uint8_t **buf, size_t *len) const = 0;
+
+};
+class CECFrame{
+	public:
+	static CECFrame& getInstance()
+    	{
+                static CECFrame instance;
+       	        return instance;
+	}
+
+	enum {
+		MAX_LENGTH = 128,
+	};
+	CECFrameImpl* impl;
+	void getBuffer(const uint8_t **buf, size_t *len) const{
+		return getInstance().impl->getBuffer(buf, len);
+	}
+	CECFrame(const uint8_t *buf = NULL, uint16_t len = 0){
+		
+	}
+	private:
```

Comment:
please don't copy -paste unnecessary code. e.g. here `CECFrame::getInstance()`  is not used, class `CECFrameImpl` is not used, MAX_LENGTH is not used, `len_` is not used. Reading further, `LogicalAddress::getInstance()` is not used, `LogicalAddressImpl` is not used, etc.